lms/djangoapps/teams/tests/test_views.py

```diff
@@ -0,0 +1,454 @@
+"""Tests for the teams API at the HTTP request level."""
+# pylint: disable=maybe-no-member
+import json
+
+import ddt
+
+from django.core.urlresolvers import reverse
+from rest_framework.test import APITestCase
+
+from student.tests.factories import UserFactory
+from student.models import CourseEnrollment
+from xmodule.modulestore.tests.factories import CourseFactory
+from .factories import CourseTeamFactory
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+
+
+@ddt.ddt
+class TestTeamAPI(APITestCase, ModuleStoreTestCase):
+    """
+    Tests of the Team API
+    """
+
+    test_password = 'password'
+
+    topics_count = 10
+
+    def setUp(self):
+        super(TestTeamAPI, self).setUp()
+
+        teams_configuration = {
+            'topics':
+            [
+                {
+                    'id': 'topic_{}'.format(i),
+                    'name': 'Topic {}'.format(i),
+                    'description': 'Description for topic {}.'.format(i)
+                } for i in range(self.topics_count)
+            ]
+        }
+
+        self.test_course_1 = CourseFactory.create(
+            org='TestX',
+            course='TS101',
+            display_name='Test Course',
+            teams_configuration=teams_configuration
+        )
+        self.test_course_2 = CourseFactory.create(org='MIT', course='6.002x', display_name='Circuits')
+
+        self.student_user = UserFactory.create(password=self.test_password)
+        self.student_user_enrolled = UserFactory.create(password=self.test_password)
+        self.student_user_not_active = UserFactory.create(password=self.test_password, is_active=False)
+        self.staff_user = UserFactory.create(password=self.test_password, is_staff=True)
+
+        self.test_team_1 = CourseTeamFactory.create(
+            name='solar team',
+            course_id=self.test_course_1.id,
+            topic_id='renewable'
+        )
+        self.test_team_2 = CourseTeamFactory.create(name='Wind Team', course_id=self.test_course_2.id)
+        self.test_team_3 = CourseTeamFactory.create(name='Nuclear Team', course_id=self.test_course_1.id)
+        self.test_team_4 = CourseTeamFactory.create(name='coal team', course_id=self.test_course_2.id, is_active=False)
```

Comment:
I'll capitalize it. For the record, 'solar team' is intentionally lower case to test case insensitivity in team name ordering. I'll add a comment explaining this.